packages/wrangler/src/__tests__/configuration.test.ts

```diff
@@ -852,6 +852,53 @@ describe("normalizeAndValidateConfig()", () => {
 		        `);
 			});
 		});
+
+		it("should warn on unsafe binding metadata usage", () => {
+			const expectedConfig: RawConfig = {
+				unsafe: {
+					bindings: [
+						{
+							type: "metadata",
+							name: "METADATA",
+						},
+					],
+				},
+			};
+
+			const { config, diagnostics } = normalizeAndValidateConfig(
+				expectedConfig,
+				"project/wrangler.toml",
+				{ env: undefined }
+			);
+
+			expect(config).toEqual(
+				expect.objectContaining({
+					unsafe: {
+						bindings: [
+							{
+								type: "metadata",
+								name: "METADATA",
+							},
+						],
+					},
+				})
+			);
+			expect(diagnostics.hasErrors()).toBe(false);
+			expect(diagnostics.hasWarnings()).toBe(true);
+
+			console.log(diagnostics.renderWarnings());
+			expect(diagnostics.renderWarnings()).toMatchInlineSnapshot(`
+				"Processing project${path.sep}wrangler.toml configuration:
+				  - \\"unsafe\\" fields are experimental and may change or break at any time.
+				  - \\"unsafe.bindings[0]\\": {\\"type\\":\\"metadata\\",\\"name\\":\\"METADATA\\"}
+				    - The \\"deployment\\" object in the metadata binding is now deprecated. Please switch using the stable version_metadata binding for access to version specific fields:
```

Comment:
Technically, it's in beta.

```suggestion
				    - The \\"deployment\\" object in the metadata binding is now deprecated. Please switch using the new version_metadata binding for access to version specific fields:
```